File: c++.vector.c
2.5.3 vector 容器嵌套容器
#include<iostream>
using namespace std;
#include<string>
#include<vector>
void test01(){
	vector<vector<int>>v;
	//创建小容器
	vector<int>v1;
	vector<int>v2;
	vector<int>v3; 
	vector<int>v4;
	for (int i = 0; i < 4; i++){
		v1.push_back(i + 1);
		v2.push_back(i + 2);
		v3.push_back(i + 3);
		v4.push_back(i + 4);
	}
	//将小容器插入到大的容器中
	v.push_back(v1);
	v.push_back(v2);
	v.push_back(v3);
	v.push_back(v4);
	//通过大容器，把所有数据遍历一遍
	for (vector<vector<int>>::iterator it = v.begin(); it != v.end(); it++){
		//(*it)---容器vector<int>
		for (vector<int>::iterator vit = (*it).begin(); vit != (*it).end(); vit++){
			cout << *vit << " ";
		}
		cout << endl;
	}
}
int main(){
	test01();
	system("pause");
	return 0;
}
#include<iostream>
using namespace std;
#include<string>
//string的构造函数
//string();    				//创建一个空的字符串 例如: string str;
//string(const char* s); 	        //使用字符串s初始化
//string(const string& str);     //使用一个string对象初始化另一个string对象
//string(int n, char c);          //使用n个字符c初始化 


void test01(){
	string s1;//默认构造
	const char *str= "hello world";
	string s2(str);
	cout << "s2=" << s2 << endl;
	string s3(s2);
	cout << "s3=" << s3 << endl;
	string s4(10, 'a');
	cout << "s4=" << s4 << endl;
}
int main(){
	test01();
	system("pause");
	return 0;
}

File: c++.vector.m
//3.2.4  vector容量和大小
//功能描述：
//对vector容器的容量和大小操作
//函数原型：
//empty();                             //判断容器是否为空
//capacity();                       //容器的容量
//size();                               //返回容器中元素的个数
//resize(int num);              //重新指定容器的长度为num，若容器变长，则以默认值填充新位置。
//​					      //如果容器变短，则末尾超出容器长度的元素被删ch除
//resize(int num, elem);  //重新指定容器的长度为num，若容器变长，则以elem值填充新位置。
​				              //如果容器变短，则末尾超出容器长度的元素被删除
#include<iostream>
using namespace std;
#include<vector>
void printVector(vector<int>&v){
	for (vector<int>::iterator it = v.begin(); it != v.end(); it++){
		cout << *it << " ";
	}
	cout << endl;
}
void test01(){
	vector<int>v1;
	for (int i = 0; i < 10; i++){
		v1.push_back(i);
	}
	printVector(v1);
	if (v1.empty()){
		cout << "v1为空" << endl;
	}
	else{
		cout << "v1的容量为：" << v1.capacity() << endl;
		cout << "v1的大小为：" << v1.size() << endl;
	}
	//重新指定大小
	v1.resize(15,100);//利用重载版本，可以指定默认填充值，参数2 
	printVector(v1);//如果重新指定的比原来长，就默认0从新填充
	v1.resize(5);
	printVector(v1);//如果重新指定的比原来短，超出部分会删除掉
}
int main(){
	test01();
	system("pause");
	return 0;
}
//总结：
//判断是否为空-- - empty
//返回元素个数-- - size
//返回容器容量-- - capacity
//重新指定大小-- - resize
